File: implementations/php/src/postgres-bulk.php
<?php

include "hashes.php";

try {
    $conn = new PDO("pgsql:host=127.0.0.1;port=5432;dbname=hashes", "root", "root123");
    $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

    $conn->exec(<<<HEREA
            DROP TABLE IF EXISTS tb_php_bulk;
            CREATE TABLE tb_php_bulk
            (id SERIAL PRIMARY KEY,
            hash INTEGER NULL);
            HEREA);

    $query = "INSERT INTO tb_php_bulk(hash) VALUES ";

    for ($i = 1; $i <= 10000; $i++) {
        $hash = hash32($i);
        $query .= sprintf("(%d),", $hash);
    }

    $query = rtrim($query, ",");

    $conn->beginTransaction();
    $conn->exec($query);
    $conn->commit();
} catch (\Throwable $th) {
    echo $th->getMessage() . "\n";
} finally {
    $conn = null;
}

File: implementations/py/src/postgres-bulk.py
from hashes import hash32
from psycopg2 import connect

try:
    conn=None
    with connect(
        host="127.0.0.1",
        user="root",
        password="root123",
        database="hashes",
    ) as conn:
        with conn.cursor() as cur:
            cur.execute(
                """DROP TABLE IF EXISTS tb_py_bulk;
                CREATE TABLE tb_py_bulk
                (id SERIAL PRIMARY KEY,
                hash INTEGER NOT NULL);"""
            )
            conn.commit()

            inserts = []
            for i in range(1, 10_001):
                hash = hash32(i)
                inserts.append(f"({hash})")

            query = (
                "BEGIN;\nINSERT INTO tb_py_bulk(hash) VALUES "
                + ",".join(inserts)
                + ";\nCOMMIT;"
            )
            cur.execute(query)

except Exception as e:
    print(e)

